File: solutions/cpp/05-iz3/code/src/main.cpp
#include <iostream>
#include <string>

int main() {
  // Flush after every std::cout / std:cerr
  std::cout << std::unitbuf;
  std::cerr << std::unitbuf;

  while (true) {
    std::cout << "$ ";

    std::string input;
    std::getline(std::cin, input);

    if (input == "exit") {
      break;
    } else if (input.substr(0, 5) == "echo ") {
      std::cout << input.substr(5) << std::endl;
    } else {
      std::cout << input << ": command not found" << std::endl;
    }
  }
}

File: solutions/csharp/05-iz3/code/src/main.cs
class Program
{
    static void Main()
    {
        while (true)
        {
            Console.Write("$ ");

            // Wait for user input
            var command = Console.ReadLine();
            if (command is null || command == "exit")
            {
                break;
            }
            else if (command == "echo")
            {
                Console.WriteLine();
            }
            else if (command.StartsWith("echo "))
            {
                Console.WriteLine(command[5..]);
            }
            else
            {
                Console.WriteLine($"{command}: command not found");
            }
        }
    }
}
